odoo/addons/base/models/ir_http.py

```diff
@@ -201,6 +202,14 @@ def _match(cls, path_info):
     def _get_public_users(cls):
         return [request.env['ir.model.data']._xmlid_to_res_model_res_id('base.public_user')[1]]
 
+    @classmethod
+    def _auth_method_bearer(cls):
+        uid = request.env.uid
+        if uid is None and request.httprequest.authorization:
+            uid = cls._auth_header(request.httprequest.authorization)
+            request.update_env(user=uid)
```

Comment:
It should use the bearer first, and only if not set use the session
```suggestion
        if request.httprequest.authorization:
            uid = cls._auth_header(request.httprequest.authorization)
            request.update_env(user=uid)
```

Reply:
we could inline the `_auth_header` here

Reply:
I will inline now that it is a separate auth method.
However, I was explicitely not checking the header when there is an uid in the session. Do you want me to always check the header first?

Reply:
> Do you want me to always check the header first?

As you are giving the header explicitly I would expect it takes precedence on the session